app/src/services/ShopService.ts

```diff
@@ -381,6 +400,12 @@ export const ShopService: ShopControllerSocket & DashboardControllerSocket = {
       console.error('Invalid date, earlier than today')
       throw new InvalidParamsError()
     }
+    const reservation = await ReservationRepository.fetchShopReservationsBySS(shopId, stylistId)
+    const availability = checkAvailablity(reservation, reservationDate).pop()
+    if (!availability || availability === undefined) {
+      console.error('Reservation unavailable')
+      throw new DuplicateModelError()
```

Comment:
maybe duplicate model is not the right error for this, maybe you can come up with a new one